src/server/utils/logging/logger.js

```diff
@@ -26,24 +26,33 @@ import baseConfig from './config/base';
 export function createLogger() {
   const useProductionConfig = !!(argv.logFormat === 'machine' || process.env.NODE_ENV !== 'development');
 
-  let transport;
+  let pinoConfig = baseConfig;
+  const transportStreams = [];
+
+  if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+    transportStreams.push(createOtelTransport({
+      grpc: true,
+      console: process.env.NODE_ENV === 'development' && argv.logFormat === 'machine',
+    }));
+    pinoConfig = deepmerge(baseConfig, otelConfig);
+  } else if (useProductionConfig) {
+    pinoConfig = deepmerge(baseConfig, productionConfig);
+  }
 
-  if (!useProductionConfig && !process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+  if (!useProductionConfig) {
     // eslint-disable-next-line global-require -- do not load development logger in production
-    transport = require('./config/development').default;
-  } else if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
-    transport = createOtelTransport();
+    transportStreams.push(require('./config/development').default);
+  }
```

Comment:
the problem with mutation, especially here, is that cases for transport streams and for pino config are different, I would convert them into functions to make it more readable, like

```js
const getTransportStreams = () => {
  const transportStreams = [];

  if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
    transportStreams.push(createOtelTransport({
      grpc: true,
      console: process.env.NODE_ENV === 'development' && argv.logFormat === 'machine',
    }));
  }

  if (!useProductionConfig) {
    // eslint-disable-next-line global-require -- do not load development logger in production
    transportStreams.push(require('./config/development').default);
  }

  return transportStreams;
}

const getPinoConfig = () => {
  if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
    return deepmerge(baseConfig, otelConfig);
  }

  if (useProductionConfig) {
    return deepmerge(baseConfig, productionConfig);
  }

  return baseConfig
}
```